sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+"""TEASER early classifier.
+
+An early classifier using a one class SVM's to determine decision safety with a
+time series classifier.
+"""
+
+__author__ = ["MatthewMiddlehurst"]
+__all__ = ["TEASER"]
+
+import copy
+
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import GridSearchCV, cross_val_predict
+from sklearn.svm import OneClassSVM
+from sklearn.utils import check_random_state
+
+from sktime.base._base import _clone_estimator
+from sktime.classification.base import BaseClassifier
+from sktime.classification.dictionary_based import MUSE, WEASEL
+from sktime.classification.feature_based import Catch22Classifier
+from sktime.utils.validation.panel import check_X
+
+
+class TEASER(BaseClassifier):
+    """Two-tier Early and Accurate Series Classifier (TEASER).
+
+    An early classifier which uses one class SVM's trained on prediction probabilities
+    to determine whether an early prediction is safe or not.
+
+    Overview:
+        Build n classifiers, where n is the number of classification_points.
+        For each classifier, train a one class svm used to determine prediction safety
+        at that series length.
+        Tune the number of consecutive safe svm predictions required to consider the
+        prediction safe.
+
+        While a prediction is still deemed unsafe:
+            Make a prediction using the series length at classification point i.
+            Decide whether the predcition is safe or not using decide_prediction_safety.
+
+    Parameters
+    ----------
+    estimator: sktime classifier, default=None
```

Comment:
As the classifier is parameterized, it might be worth parameterizing the anomaly detector, too. It is a One-Class-SVM but it doesn't have to be. It might as well be any other method available in sklearn such as: IsolationForests, ...

Reply:
Done.